src/main/java/Frappe.java

```diff
@@ -0,0 +1,308 @@
+                } else if (task instanceof Event) {
+                    writer.write(task.getFrom());
+                    writer.write("\n");
+                    writer.write(task.getTo());
+                    writer.write("\n");
+                }
+
+                writer.write("\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new FrappeFileException(FrappeFileException.SAVE_FAILED_WARNING);
+        }
+    }
+
+    private static void loadTasks() throws FrappeFileException {
+        try {
+            File saveFile = new File(saveFilePath);
+            Scanner in = new Scanner(saveFile);
+            in.useDelimiter("\n\n");
+            while (in.hasNext()) {
+                String input = in.next();
+                String[] inputs = input.split("\n");
+
+                for (int i = 0; i < inputs.length; i++) {
+                    inputs[i] = inputs[i].trim();
+                }
+
+                if (inputs.length < 2) {
+                    throw new FrappeFileException(FrappeFileException.CORRUPTED_SAVE_WARNING);
+                }
+
+                Task task;
+                String name = inputs[2];
+                boolean marked;
+
+                if (inputs[1].equals("0")) {
+                    marked = false;
+                } else if (inputs[1].equals("1")) {
+                    marked = true;
+                } else {
+                    throw new FrappeFileException(FrappeFileException.CORRUPTED_SAVE_WARNING);
+                }
+
+                switch (inputs[0]) {
+                case "[T]":
+                    if (inputs.length != 3) {
+                        throw new FrappeFileException(FrappeFileException.CORRUPTED_SAVE_WARNING);
+                    }
+                    task = new Todo(name);
+                    break;
+
+                case "[D]":
+                    if (inputs.length != 4) {
+                        throw new FrappeFileException(FrappeFileException.CORRUPTED_SAVE_WARNING);
+                    }
+                    String by = inputs[3];
+                    task = new Deadline(name, by);
+                    break;
+
+                case "[E]":
+                    if (inputs.length != 5) {
+                        throw new FrappeFileException(FrappeFileException.CORRUPTED_SAVE_WARNING);
+                    }
+                    String from = inputs[3];
+                    String to = inputs[4];
+                    task = new Event(name, from, to);
+                    break;
+
+                default:
+                    throw new FrappeFileException(FrappeFileException.CORRUPTED_SAVE_WARNING);
+                }
+
+                task.setDone(marked);
+                tasks.add(task);
+            }
+        } catch (FileNotFoundException e) {
+            throw new FrappeFileException(FrappeFileException.NO_SAVE_WARNING);
+        }
+    }
+
+    public static void main(String[] args) {
```

Comment:
I believe you can SLAP harder here, `processInput`, `Scanner`,  `split` etc. and the rest of the function calls are of varying levels of abstraction. More notably the lower levels of parsing should be abstracted out